src/expr/src/relation/mod.rs

```diff
@@ -1163,12 +1163,16 @@ pub struct AggregateExpr {
     pub expr: ScalarExpr,
     /// Should the aggregation be applied only to distinct results in each group.
     pub distinct: bool,
+    /// Can the aggregation return null
+    pub nullable: bool,
```

Comment:
I'm not sure I understand the role of this. The `AggregateFunc::Count` variant cannot return null, but all others can.

Reply:
If you do a partial pushdown of a `count(something)`, the inner aggregation will be `count(something)`  and the outer aggregation will be `sum(the_count_column)`. The outer aggregation is not nullable and needs to be marked as such. Otherwise, Materialize can crash at dataflow render time if the reduce is inside the `value` of a `Let` statement, because the `RelationType` of what is rendered will not match the `RelationType` in the `Get` statement inside the `body` of the `Let` statement.

Reply:
I see. I think this is probably a bug in `AggregateFunc::output_type` where it ignores nullability information in the the types of its inputs. If that is right, and a fix happens there, is it correct that this is not required?

Reply:
Not quite because the nullability of an aggregation like sum depends not only on the nullability of the types of its inputs but also whether or not the input can have zero rows at some point. `sum(the_count_column)` is not nullable because its input is not nullable _and_ guaranteed to have at least one row.

Reply:
In `expr::RelationExpr` I think we only produce `Datum::Null` as output if there exist a null in the input. On the other hand, `sql::RelationExpr` may do what you describe. We had a side convo in #sql and I think we agreed that conflating these two isn't right. I have a PR that de-conflates them, but some tests are failing that I haven't checked out yet (#5480).